torchmultimodal/modules/encoders/albef_text_encoder.py

```diff
@@ -0,0 +1,263 @@
+        intermediate_size (int): Dimensionality of the “intermediate” (i.e., feed-forward) layer in the Transformer encoder.
+        max_position_embeddings (int): The maximum sequence length that this model might ever be used with.
+        type_vocab_size (int): The vocabulary size of the token_type_ids.
+        pad_token_id (int): The embedding for pad_token_id is not updated during training.
+        layer_norm_eps (float): The epsilon used by the layer normalization layers.
+    Inputs:
+        input_ids (Tensor of size (batch_size, sequence_length)): Indices of input sequence tokens in the vocabulary.
+        attention_mask (Tensor of shape (batch_size, sequence_length)): Mask to avoid performing attention on padding token indices.
+            Mask values selected in [0, 1]: 1 for tokens that are NOT MASKED, 0 for MASKED tokens.
+    """
+
+    def __init__(
+        self,
+        vocab_size: int = 30522,
+        hidden_size: int = 768,
+        num_hidden_layers: int = 6,
+        num_attention_heads: int = 12,
+        intermediate_size: int = 3072,
+        max_position_embeddings: int = 512,
+        type_vocab_size: int = 2,
+        pad_token_id: int = 0,
+        layer_norm_eps: float = 1e-12,
+    ) -> None:
+        super().__init__()
+
+        self.embeddings = ALBEFTextEmbeddings(
+            vocab_size,
+            hidden_size,
+            pad_token_id,
+            max_position_embeddings,
+            type_vocab_size,
+            layer_norm_eps,
+        )
+        self.encoder = ALBEFTransformerEncoder(
+            hidden_size,
+            intermediate_size,
+            num_attention_heads,
+            num_hidden_layers,
+            layer_norm_eps,
+        )
+
+    def forward(
+        self,
+        input_ids: Tensor = None,
+        attention_mask: Optional[Tensor] = None,
+    ) -> Tensor:
+        input_shape = input_ids.size()
+        device = input_ids.device
+        if attention_mask is None:
+            attention_mask = torch.ones(input_shape, device=device)
+
+        extended_attention_mask = get_extended_attention_mask(attention_mask)
+
+        embedding_output = self.embeddings(input_ids)
+        encoder_outputs = self.encoder(embedding_output, extended_attention_mask)
+
+        return encoder_outputs
+
+
+class ALBEFTextEmbeddings(nn.Module):
+    def __init__(
+        self,
+        vocab_size: int,
+        hidden_size: int,
+        padding_idx: int,
+        max_position_embeddings: int,
+        type_vocab_size: int,
+        layer_norm_eps: float,
+    ) -> None:
+        super().__init__()
+        self.word_embeddings = nn.Embedding(vocab_size, hidden_size, padding_idx)
+        self.position_embeddings = nn.Embedding(max_position_embeddings, hidden_size)
+        self.token_type_embeddings = nn.Embedding(type_vocab_size, hidden_size)
+        self.LayerNorm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
```

Comment:
I would change the name of this slighty so it's not too similar to `nn.LayerNorm`
```suggestion
        self.layer_norm = nn.LayerNorm(hidden_size, eps=layer_norm_eps)
```